### ISSUE TITLE
rsync -e 'tsh ssh' … : ERROR: unknown long flag '--server'

### ISSUE THREAD

**Issue by Mindstan** [gravitational/teleport#16044]
Expected behavior:

`tsh ssh` should work as a drop-in replacement for `ssh` for the command `rsync` as [stated in the support](https://support.goteleport.com/hc/en-us/articles/4649132890131-How-to-use-teleport-with-other-applications)

Current behavior:

It looks like `tsh ssh` interpretes remote `rsync` flags as its own, and fails because they do not exist.

Bug details:
- Teleport version: Teleport v10.1.9 git:v10.1.9-0-g8f6f6cc go1.18.3 on Ubuntu 20.04.5 x86-64, `rsync` 3.1.3
- Recreation steps
    ```bash
    tsh login --proxy my.proxy --auth MyAuth
    rsync -e 'tsh ssh' -aPh --debug=CMD my_file.txt user@machine:my_file.txt 
    ```
- Debug logs
   ```
    opening connection using: tsh ssh -l user machine rsync --server -vlogDtpre.iLsfxC --partial . "~/my_file.txt" (11 args)
    tsh: error: unknown long flag '--server'
    usage: tsh ssh [<flags>] <[user@]host> [<command>...]
    
    Run shell or execute a command on a remote SSH node
    
    Flags:
      -l, --login                    Remote host login
          --proxy                    SSH proxy address
          --user                     SSH proxy user [gaetan]
          --ttl                      Minutes to live for a SSH session
      -i, --identity                 Identity file
          --cert-format              SSH certificate format
          --insecure                 Do not verify server's certificate and host name. Use only in test environments
          --auth                     Specify the name of authentication connector to use.
          --skip-version-check       Skip version checking between server and client.
      -d, --debug                    Verbose logging to stdout
      -k, --add-keys-to-agent        Controls how keys are handled. Valid values are [auto no yes only].
          --enable-escape-sequences  Enable support for SSH escape sequences. Type '~?' during an SSH session to list supported sequences. Default is enabled.
          --bind-addr                Override host:port used when opening a browser for cluster logins
          --mfa-mode                 Preferred mode for MFA and Passwordless assertions (auto, cross-platform, platform, otp)
      -J, --jumphost                 SSH jumphost
      -p, --port                     SSH port on a remote host
      -A, --forward-agent            Forward agent to target node
      -L, --forward                  Forward localhost connections to remote server
      -D, --dynamic-forward          Forward localhost connections to remote server using SOCKS5
          --local                    Execute command on localhost after connecting to SSH node
      -t, --tty                      Allocate TTY
      -c, --cluster                  Specify the Teleport cluster to connect
      -o, --option                   OpenSSH options in the format used in the configuration file
      -N, --no-remote-exec           Don't execute remote command, useful for port forwarding
      -X, --x11-untrusted            Requests untrusted (secure) X11 forwarding for this session
      -Y, --x11-trusted              Requests trusted (insecure) X11 forwarding for this session. This can make your local displays vulnerable to attacks, use with caution
          --x11-untrusted-timeout    Sets a timeout for untrusted X11 forwarding, after which the client will reject any forwarding requests from the server
          --participant-req          Displays a verbose list of required participants in a moderated session.
          --request-reason           Reason for requesting access
          --disable-access-request   Disable automatic resource access requests
    
    Args:
      <[user@]host>  Remote hostname and the login to use
      [<command>]    Command to execute on a remote host
    
    Aliases:
    
    ERROR: unknown long flag '--server'
    
    rsync: connection unexpectedly closed (0 bytes received so far) [sender]
    rsync error: error in rsync protocol data stream (code 12) at io.c(235) [sender=3.1.3]
    ```

**Comment by greedy52** [comment]
Same error as #15965. Will fix together

**Comment by greedy52** [comment]
Closed by #16058
